rust/agents/scraper/src/db/txn.rs

```diff
@@ -60,26 +63,30 @@ impl ScraperDb {
                 Ok(transaction::ActiveModel {
                     id: NotSet,
                     block_id: Unchanged(txn.block_id),
-                    gas_limit: Set(as_f64(txn.gas_limit)),
-                    max_priority_fee_per_gas: Set(txn.max_priority_fee_per_gas.map(as_f64)),
-                    hash: Unchanged(format_h256(&txn.hash)),
+                    gas_limit: Set(u256_to_decimal(txn.gas_limit)),
+                    max_priority_fee_per_gas: Set(txn
+                        .max_priority_fee_per_gas
+                        .map(u256_to_decimal)),
+                    hash: Unchanged(h256_to_bytes(&txn.hash)),
                     time_created: Set(date_time::now()),
-                    gas_used: Set(as_f64(receipt.gas_used)),
-                    gas_price: Set(txn.gas_price.map(as_f64)),
-                    effective_gas_price: Set(receipt.effective_gas_price.map(as_f64)),
+                    gas_used: Set(u256_to_decimal(receipt.gas_used)),
+                    gas_price: Set(txn.gas_price.map(u256_to_decimal)),
+                    effective_gas_price: Set(receipt.effective_gas_price.map(u256_to_decimal)),
                     nonce: Set(txn.nonce as i64),
-                    sender: Set(format_h256(&txn.sender)),
-                    recipient: Set(txn.recipient.as_ref().map(format_h256)),
-                    max_fee_per_gas: Set(txn.max_fee_per_gas.map(as_f64)),
-                    cumulative_gas_used: Set(as_f64(receipt.cumulative_gas_used)),
+                    sender: Set(address_to_bytes(&txn.sender)),
+                    recipient: Set(txn.recipient.as_ref().map(address_to_bytes)),
+                    max_fee_per_gas: Set(txn.max_fee_per_gas.map(u256_to_decimal)),
+                    cumulative_gas_used: Set(u256_to_decimal(receipt.cumulative_gas_used)),
                 })
             })
             .collect::<Result<Vec<_>>>()?;
 
         debug_assert!(!models.is_empty());
+        let id_offset = models.len() as i64 - 1;
+        debug!(txns = models.len(), "Writing txns to database");
         trace!(?models, "Writing txns to database");
-        // this is actually the ID that was first inserted for postgres
-        let first_id = Insert::many(models).exec(&self.0).await?.last_insert_id;
+        let first_id = Insert::many(models).exec(&self.0).await?.last_insert_id - id_offset;
```

Comment:
Yep, this is because they made `last_insert_id` actually the last id and not the first which it used to be for postgres. I assume this was a mysqlism that they had not properly handled for the postgres case.